torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -100,6 +100,36 @@ namespace {
 namespace onnx_torch = ::torch::onnx;
 namespace onnx = ::ONNX_NAMESPACE;
 
+c10::ShapeSymbol ONNXDimToShapeSymbol(onnx::TensorShapeProto_Dimension dim, SymbolDimMap& symbol_map){
+  if (dim.has_dim_value()) {
+    return c10::ShapeSymbol::fromStaticSize(dim.dim_value());
+  }
+  c10::optional<c10::ShapeSymbol> sym = c10::nullopt;
+  if (dim.has_dim_param()) {
+    // A specific dim param is produced.
+    // Search if this is already known,
+    // and assign the same Symbol.
+    GRAPH_UPDATE("Got dim_param:", dim.dim_param());
+    for (const auto& pair : symbol_map) {
+      if (pair.second == dim.dim_param()) {
+        sym = pair.first;
+        break;
+      }
+    }
+    if (!sym) {
+      sym = c10::ShapeSymbol::newSymbol();
+      symbol_map[sym.value()] = dim.dim_param();
+    }
+  } else {
```

Comment:
Can we get rid of this `else` clause and combine this with the code that handles `if (!sym)`?
Because:
1. We know that if the if on line 108 evaluates to false, then `sym` will be null.
2. Also if 108 evaluates to false, then `dim.dim_param() == ""`.

So I think we can combine the if statement on line 119 and this else block.
And I think that'd be clearer, because there's one fewer case to understand.

Reply:
Brilliant idea! I have eliminated these 3 if conditions here to 2 if conditions. Please review it again. Thanks.